// @keystonejs/fields/dist/Controller-66702d8d.cjs.js
'use strict';

function _interopDefault(ex) {
  return ex && typeof ex === 'object' && 'default' in ex ? ex['default'] : ex;
}

var _defineProperty = _interopDefault(require("@babel/runtime/helpers/defineProperty"));

require("@babel/runtime/helpers/objectWithoutProperties");

require('inflection');

require('@keystonejs/access-control');

var fields = require('./index-a9a28ae7.cjs.js');

require('date-fns/parse');

require('date-fns/format');

require('@keystonejs/adapter-mongoose');

require('@keystonejs/adapter-knex');

require('mongoose');

require('pluralize');

require('@keystonejs/field-content/Block');

require('@keystonejs/field-content/blocks');

require('lodash.groupby');

require('p-settle');

require('@keystonejs/utils');

require('apollo-errors');

require('graphql');

require('graphql/language');

require('luxon');

require('node-fetch');

require('dumb-passwords');

require('@sindresorhus/slugify');

require('cuid');

require('unsplash-js');

require('query-string');

require("@babel/runtime/helpers/classCallCheck");

require('lodash.isequal');

require('../Controller/dist/fields.cjs.js');

var Controller = require('./Controller-1e11bb72.cjs.js');

function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);

  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);
    if (enumerableOnly) symbols = symbols.filter(function (sym) {
      return Object.getOwnPropertyDescriptor(object, sym).enumerable;
    });
    keys.push.apply(keys, symbols);
  }

  return keys;
}

function _objectSpread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};

    if (i % 2) {
      ownKeys(Object(source), true).forEach(function (key) {
        _defineProperty(target, key, source[key]);
      });
    } else if (Object.getOwnPropertyDescriptors) {
      Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
      ownKeys(Object(source)).forEach(function (key) {
        Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
      });
    }
  }

  return target;
}

class FileController extends Controller.default {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "getQueryFragment", () => "\n    ".concat(this.path, " {\n      ").concat(fields.queryFragment, "\n    }\n  "));

    _defineProperty(this, "deserialize", data => {
      const oEmbed = data[this.path];

      if (!oEmbed || !oEmbed.originalUrl) {
        // Nothing set, so force to null
        return null;
      }

      return {
        originalUrl: oEmbed.originalUrl,
        preview: _objectSpread({
          // Ensure the .html property is always set
          html: ''
        }, oEmbed, {
          // Attempt to derive a useful title
          title: oEmbed.title || oEmbed.author && oEmbed.author.name || oEmbed.provider && oEmbed.provider.name || ''
        })
      };
    });

    _defineProperty(this, "serialize", data => {
      const {
        path
      } = this; // We only send the URL itself to the mutation

      return data[path] && data[path].originalUrl || null;
    });
  }

}

exports.default = FileController;
